components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1329,4 +1334,32 @@ void ConversationDriver::SendFeedback(
                                   : std::nullopt,
                               std::move(on_complete));
 }
+
+void ConversationDriver::ModifyConversation(uint32_t turn_index,
+                                            const std::string& new_text) {
+  if (turn_index >= chat_history_.size()) {
+    return;
+  }
+
+  auto& turn = chat_history_.at(turn_index);
+  if (turn->character_type == CharacterType::ASSISTANT) {  // not supported yet
+    return;
+  }
+
+  std::string sanitized_input = new_text;
+  engine_->SanitizeInput(sanitized_input);
+  if (sanitized_input.empty() || sanitized_input == turn->text) {
+    return;
+  }
+
+  turn->edits.push_back(turn.Clone());
+  turn->text = sanitized_input;
+  turn->last_edited_time = base::Time::Now();
+
+  // Modifying human turn, drop anything after this turn_index and resubmit.
+  auto new_turn = std::move(chat_history_.at(turn_index));
+  chat_history_.erase(chat_history_.begin() + turn_index, chat_history_.end());
```

Comment:
Originally I didn't do so since it would be called in SubmitHumanConversationEntry, but I'm fine having another one here after we clear it. Added in https://github.com/brave/brave-core/pull/24558/commits/26769e4103cbe1d3d64a756c8fa49a214643e1ca.